src/test/java/kitchenpos/application/MenuGroupServiceTest.kt

```diff
@@ -0,0 +1,64 @@
+package kitchenpos.application
+
+import io.kotest.assertions.throwables.shouldThrowExactly
+import io.kotest.matchers.shouldBe
+import io.mockk.every
+import io.mockk.impl.annotations.InjectMockKs
+import io.mockk.impl.annotations.MockK
+import io.mockk.junit5.MockKExtension
+import kitchenpos.domain.MenuGroup
+import kitchenpos.domain.MenuGroupRepository
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.NullAndEmptySource
+import java.util.*
+
+@ExtendWith(MockKExtension::class)
+internal class MenuGroupServiceTest {
+    @MockK
+    private lateinit var menuGroupRepository: MenuGroupRepository
+
+    @InjectMockKs
+    private lateinit var menuGroupService: MenuGroupService
+
+    @Nested
+    inner class `메뉴 그룹 생성 테스트` {
+        @DisplayName("메뉴 그룹에 이름이 null 또는 길이가 0일 경우, IllegalArgumentException 예외 처리를 한다.")
+        @ParameterizedTest
+        @NullAndEmptySource
+        fun test1(name: String?) {
+            // given
+            val request = MenuGroup().apply {
+                this.id = UUID.randomUUID()
+                this.name = name
+            }
```

Comment:
Test Fixture를 반환하는 별도의 메서드를 별도로 정의하여 가독성과 재사용성을 높여보면 어떨까요?
```suggestion
            val request = createMenuGroup(name)
```

Reply:
몇 개 테스트들을 Fixture로 만들면서 느낀점 입니다.

해당 요구사항에서 뭔가 강제로 fixture를 집어넣는 느낌이 납니다.
재사용성은 향상되나, 가독성 측면에서 이게 정말 좋아졌는가? 라는 의문이 들긴 합니다.

실제로 정말 매 테스트마다 반복된 데이터를 여러 개 저장해야 한다면 Fixture 사용이 좋다고 생각되지만, 해당 테스트들 처럼, 각 테스트 마다 한 가지의 상태만 차이가 나는 테스트들이 대부분일 때, 오히려 Fixture 사용이 가독성을 떨어트린다고 생각합니다.
given 값을 보며, 해당 테스트 의도를 정확히 알 수 있다고 생각합니다.

-> 제가 Fixture 만드는 방법이 잘못돼 이렇게 느끼는 것 같기도 합니다. 학성님의 예시가 궁금합니다!

Reply:
> given 값을 보면, 해당 테스트 의도를 정확히 알 수 있다고 생각합니다.
> -> 제가 Fixture 만드는 방법이 잘못돼 이렇게 느끼는 것 같기도 합니다. 학성님의 예시가 궁금합니다!

재권님 말씀대로 테스트코드는 테스트 의도를 정확하고 쉽게 알 수 있도록 만들어야 한다는 의견에 전적으로 동의합니다. 😄 
제가 Fixture를 생성해주는 공통 메서드를 제안 드린 의도는 다음과 같습니다.
일단 테스트 메서드마다 fixture 객체를 생성하는 패턴이 비슷해 보였고 
특히 반복되는 코드를 메서드로 따로 분리하면 테스트 코드가 더 줄어들 것으로 생각했습니다.
혹시나 향후 Fixture에 속성이 추가되더라도 모든 테스트 메서드마다 따라다니면서 추가하는 것이 아닌 
공통메서드에만 추가하면 되는 이점도 있구요.
어떤 Fixture를 만드는지는 메서드에 전달되는 파라미터값으로 의도를 파악할 수 있을거라 생각했습니다.
MenuServiceTest를 가지고 제가 생각한 예시를 든다면 아래 처럼 될 것 같습니다.  

```
    fun test1() {
          천원짜리연필 = createMenuRequest("연필", 1000)
          ~~~
    }

    fun test2() {
          학용품그룹의이천원짜리공책 = createMenuRequest("공책", 2000, 학용품그룹)
          ~~~
    }

    fun createMenuRequest(name: String?, price: BigDecimal?): Menu {
        return Menu().apply {
            this.id = UUID.randomUUID()
            this.name = name
            this.price = price
        }
    }

    fun createMenuRequest(name: String?, price: BigDecimal?, menuGroup: MenuGroup): Menu {
        return createMenuRequest(name, price).apply {
            this.menuGroupId = UUID.randomUUID()
            this.menuGroup = menuGroup
        }
    }
```

Reply:
아하 학성님의 코드와 금일 수업 진행 내용 기반으로 의도를 이해하였습니다!
특히 코틀린에서는 디폴트 값을 정의할 수 있기에 자바에 비해 더욱 편하게 할 수 있을 것 같습니다.

적용하도록 하겠습니다!